discovery/gossiper.go

```diff
@@ -2351,91 +2349,112 @@ func (d *AuthenticatedGossiper) processNetworkAnnouncement(nMsg *networkMsg) []n
 	}
 }
 
-// sendAnnSigReliably will try to send the provided local AnnounceSignatures
-// to the remote peer, waiting for it to come online if necessary. This
-// method returns after adding the message to persistent storage, such
-// that the caller knows that the message will be delivered at one point.
-func (d *AuthenticatedGossiper) sendAnnSigReliably(
-	msg *lnwire.AnnounceSignatures, remotePeer *btcec.PublicKey) error {
-
-	// We first add this message to the database, such that in case
-	// we do not succeed in sending it to the peer, we'll fetch it
-	// from the DB next time we start, and retry. We use the peer ID
-	// + shortChannelID as key, as there possibly is more than one
-	// channel opening in progress to the same peer.
-	var key [41]byte
-	copy(key[:33], remotePeer.SerializeCompressed())
-	binary.BigEndian.PutUint64(key[33:], msg.ShortChannelID.ToUint64())
+// sendMsgReliably attempts to send the message to the peer.
+func (d *AuthenticatedGossiper) sendMsgReliably(msg lnwire.Message,
+	peerKey *btcec.PublicKey) error {
 
+	// First, we'll add the message to the database to ensure we can resend
+	// it upon restarts if the peer has yet to come back online. Each
+	// message is keyed by a tuple of the peer's public key and the message
+	// type to uniquely identify messages in the store.
+	compressedKey := peerKey.SerializeCompressed()
+	msgKey := append(compressedKey, byte(msg.MsgType()))
```

Comment:
If no migration takes place, then this breaks any unsent messages for older nodes.

Reply:
Because we capture the database key within the resend logic in `resendPendingMsgs`, I don't think a migration is needed here.

Reply:
I think it'd be worth doing a migration to make the format consistent. As @halseth suggested above, each key should be composed of unique values specific to the message type, only requiring `MsgType()` in the key if their identifiers are otherwise identical.

Reply:
Fixed.